src/webui/api/rsscontroller.cpp

```diff
@@ -139,3 +165,26 @@ void RSSController::rulesAction()
 
     setResult(jsonObj);
 }
+
+void RSSController::matchingArticlesAction()
+{
+    requireParams({"ruleName"});
+
+    const QString ruleName {params()["ruleName"]};
+    const RSS::AutoDownloadRule rule = RSS::AutoDownloader::instance()->ruleByName(ruleName);
+
+    QJsonObject jsonObj;
+    for (const QString &feedURL : rule.feedURLs()) {
+        const RSS::Feed *feed = RSS::Session::instance()->feedByURL(feedURL);
+        if (!feed) continue; // feed doesn't exist
+
+        QJsonArray matchingArticles;
+        for (const RSS::Article *article : feed->articles())
```

Comment:
According to qBittorrent Coding Guidelines you should add braces to loop body since it has several lines of code.